main.cpp

```diff
@@ -164,114 +175,530 @@ good to go!
 
 #include <iostream>
 
+struct IntType;
+struct DoubleType;
+struct FloatType;
+
 struct FloatType
 {
-    float add(float lhs, float rhs);
-    float subtract(float lhs, float rhs);
-    float multiply(float lhs, float rhs);
-    float divide(float lhs, float rhs);
+    float* value;
+    float value_;
+    FloatType(float value_) : value(new float (value_)){}
+    ~FloatType()
```

Comment:
Re-read the instructions about what your class is supposed to own.  You have more member variables than you're supposed to.